app/controllers/api/v1/mixins/rbac_mixin.rb

```diff
@@ -134,7 +134,7 @@ def visible_request_ids_for_approver
         end
 
         def assigned_group_refs
-          Insights::API::Common::RBAC::Service.call(RBACApiClient::GroupApi) do |api|
+          Insights::API::Common::RBAC::Service.call(RBACApiClient::GroupApi, Thread.current[:rbac_extra_headers] || {}) do |api|
```

Comment:
@bzwei Shouldn't this happen only on the call where we need to fetch the emails from the groups?
Instead of using the Thread.current which can get reused for different transactions, can we be explicit on which calls are admin calls. Maybe even name them as admin calls, so its easy to track them. Plus Admin calls need those extra headers, if its not there its an error.

Reply:
My intention is to refactor `ContentService#act_as_org_admin`. All RBAC calls by default do not use service_to_service headers. They are set only through `act_as_org_admin` and get cleared immediately after.